students/ShinTran/lesson06/mailroom4.py

```diff
@@ -0,0 +1,105 @@
+'''
+Shin Tran
+Python 210
+Lesson 5 Assignment
+'''
+
+master_dict = {"Jennifer Miller": [145076.19,2],
+    "William Rodriguez": [161585.16,3],
+    "Patricia Brown": [100863.96,2]
+} # For all donations made
+
+# Prompts the user to enter an option
+def main_prompt():
+    response = input("Choose from one of 4 actions:\n\
+        1) Send a Thank You\n\
+        2) Create a Report\n\
+        3) Send letters to everyone\n\
+        4) Quit\n\
+        Please type 1, 2, 3, or 4: ")
+    return response
+
+# Takes in a user input as a parameter, enters a donation, prints a report,
+# prints list, exit, prompts again if the input is bad
+# If the user types exit it'll go back to the main prompt
+def action(user_input):
+    switch_dict = {
+        'list': list_func,
+        '1': send_thanks,
+        '2': print_report,
+        '3': send_letters,
+        '4': quit
+    }
+    try:
+        switch_dict.get(user_input)()
+    except (TypeError, ValueError):
+        print("Invalid input, {} please try again.".format(user_input))
+        action(main_prompt())
+    else:
+        action(main_prompt())
+
+# Creates a list of all the distinct donors
+def list_func():
+    name_list = [k for k, v in master_dict.items()]
+    for name in name_list:
+        print(name)
+    return name_list
```

Comment:
your function should either do action - like `print` - or retrieve data

also, to get dictionary keys it's just `list(master_dict)`

Reply:
I know there's a more readable/efficient way to get the keys, but I used comprehensions there intentionally because I thought that would be a good practice for it.

Reply:
that makes sense 👍